tests/browser/automated_soft_credits/asc_primary_and_affiliation.robot

```diff
@@ -0,0 +1,29 @@
+*** Settings ***
+
+Resource        tests/NPSP.robot
+Suite Setup     Open Test Browser
+Suite Teardown  Delete Records and Close Browser
+
+ 
+*** Test Cases ***    
+Create ASC Test for Primary and Affiliations
+    [tags]  unstable
+    &{account} =  API Create Organization Account   
+    &{contact} =  API Create Contact    Email=skristem@robot.com 
+    ${ns} =  Get NPSP Namespace Prefix
+    &{affiliation} =  API Create Secondary Affiliation    &{account}[Id]    &{contact}[Id]    ${ns}Related_Opportunity_Contact_Role__c=Soft Credit
```

Comment:
Thanks Mike, I will create a new test role

Reply:
@force2b where is "Soft Credit" the default role? I checked in a scratch org and when I add an OCR using Manage Contact Roles there is no default role and role is not required. (It would be nice to avoid extra steps to deploy a new picklist value for this test if we can just use one of the normal roles.)

Reply:
@davisagli The text 'Soft Credit' is the default, but it's not an included OCR value. We include it in the trial, not packaged. So as soon as you edit in NPSP Settings, it goes away. But this way it will still create the OCR and assign the value without the user having to go to NPSP Settings to define something first. Same thing with "Donor" and "Household Member" too.

Reply:
There is a `ContactRole.standardValueSet` file in the `upackaged/config/managed/` path that creates all of these roles. If that metadata isn't deployed, then the default OCR Roles that show up in an org are automatically created by Salesforce based on the data uploaded. For example, if test data with a Role of "Soft Credit" is uploaded Salesforce will create an inactive role for "Soft Credit". I created the `ContactRole.standardValueSet` file a while ago to ensure that this metadata gets created in scratch orgs, though I thought it was deployed to all scratch orgs, not just managed scratch orgs.

Reply:
Thanks @judisohn and @force2b -- I missed that we're talking about OCRs created automatically for the primary contact, as opposed to OCRs added manually. The automatic OCR does get a default of Soft Credit in my dev org.

Reply:
True. Apex (or imported data) can assign any value to the picklist field. However, if editing or creating an OCR (or Partial Soft Credit) record in the UI it will only show active picklist values for the OpportunityContactRole.RoleName field

Reply:
@skristem I think this test can be written to use the existing picklist values for Related_Opportunity_Contact_Role__c rather than needing to deploy a new one just for this test. Instead of creating a single affilition with Related_Opportunity_Contact_Role__c=Soft Credit, it can create two: one with the Related OCR Role set to Soft Credit and one with it set to Solicitor. Then run the batch process and verify that the 2 OCRs were created with the correct roles, but that only the one with Role=Soft Credit has a soft credit value. This would be similar to the test in asc_relationships.robot for OCRs created from relationships.

Reply:
thanks David

Reply:
@davisagli and @force2b can u plz check the updated test case